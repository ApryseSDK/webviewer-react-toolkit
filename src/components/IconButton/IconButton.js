'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.IconButton = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var Button_1 = require('../Button');
exports.IconButton = (0, react_1.forwardRef)(function(_a, ref) {
  var children = _a.children,
    className = _a.className,
    props = tslib_1.__rest(_a, ['children', 'className']);
  var iconButtonClass = (0, classnames_1.default)(
    'ui__base ui__iconButton',
    className,
  );
  return react_1.default.createElement(
    Button_1.Button,
    tslib_1.__assign({}, props, {
      className: iconButtonClass,
      buttonStyle: 'borderless',
      ref: ref,
    }),
    children,
  );
});
//# sourceMappingURL=IconButton.js.map
