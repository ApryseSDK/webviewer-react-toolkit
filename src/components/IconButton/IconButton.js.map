{"version":3,"file":"IconButton.js","sourceRoot":"","sources":["IconButton.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAAgE;AAChE,oCAAmC;AAUtB,QAAA,UAAU,GAAG,IAAA,kBAAU,EAAqC,UAAC,EAAiC,EAAE,GAAG;IAApC,IAAA,QAAQ,cAAA,EAAE,SAAS,eAAA,EAAK,KAAK,sBAA/B,yBAAiC,CAAF;IACvG,IAAM,eAAe,GAAG,IAAA,oBAAU,EAAC,yBAAyB,EAAE,SAAS,CAAC,CAAC;IAEzE,OAAO,CACL,8BAAC,eAAM,uBAAK,KAAK,IAAE,SAAS,EAAE,eAAe,EAAE,WAAW,EAAC,YAAY,EAAC,GAAG,EAAE,GAAG,KAC7E,QAAQ,CACF,CACV,CAAC;AACJ,CAAC,CAAC,CAAC"}