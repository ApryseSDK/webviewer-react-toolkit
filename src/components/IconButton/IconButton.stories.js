'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importDefault(require('react'));
var action_1 = require('../../storybook-helpers/action/action');
var Icon_1 = require('../Icon/Icon');
var IconButton_1 = require('../IconButton');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/IconButton',
  component: IconButton_1.IconButton,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(
    IconButton_1.IconButton,
    {
      disabled: (0, addon_knobs_1.boolean)('disabled', false),
      onClick: (0, action_1.action)('onClick'),
    },
    react_1.default.createElement(Icon_1.Icon, { icon: 'Close' }),
  );
};
exports.Basic = Basic;
//# sourceMappingURL=IconButton.stories.js.map
