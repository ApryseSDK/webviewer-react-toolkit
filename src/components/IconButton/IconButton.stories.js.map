{"version":3,"file":"IconButton.stories.js","sourceRoot":"","sources":["IconButton.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAAiD;AACjD,wDAA0B;AAC1B,gEAA+D;AAC/D,qCAAoC;AACpC,4CAA2C;AAC3C,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,uBAAuB,EAAE,SAAS,EAAE,uBAAU,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAE1F,IAAM,KAAK,GAAG,cAAM,OAAA,CACzB,8BAAC,uBAAU,IAAC,QAAQ,EAAE,IAAA,qBAAO,EAAC,UAAU,EAAE,KAAK,CAAC,EAAE,OAAO,EAAE,IAAA,eAAM,EAAC,SAAS,CAAC;IAC1E,8BAAC,WAAI,IAAC,IAAI,EAAC,OAAO,GAAG,CACV,CACd,EAJ0B,CAI1B,CAAC;AAJW,QAAA,KAAK,SAIhB"}