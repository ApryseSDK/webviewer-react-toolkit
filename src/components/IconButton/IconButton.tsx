import classnames from 'classnames';
import React, { ButtonHTMLAttributes, forwardRef } from 'react';
import { Button } from '../Button';

export interface IconButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  /**
   * Defaults to 'button' instead of 'submit' to prevent accidental submissions.
   * @default "button"
   */
  type?: ButtonHTMLAttributes<HTMLButtonElement>['type'];
}

export const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(({ children, className, ...props }, ref) => {
  const iconButtonClass = classnames('ui__base ui__iconButton', className);

  return (
    <Button {...props} className={iconButtonClass} buttonStyle="borderless" ref={ref}>
      {children}
    </Button>
  );
});
