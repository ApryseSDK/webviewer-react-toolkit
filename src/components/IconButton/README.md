A wrapper for `Button` with custom styles to optimize it for icons.
