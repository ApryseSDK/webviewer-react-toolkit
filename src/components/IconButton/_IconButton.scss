.ui__button.ui__iconButton {
  width: auto;
  min-width: auto;
  padding: 0;
}
