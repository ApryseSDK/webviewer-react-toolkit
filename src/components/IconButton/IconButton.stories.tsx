import { action } from '@storybook/addon-actions';
import { boolean } from '@storybook/addon-knobs';
import React from 'react';
import { Icon } from '../Icon/Icon';
import { IconButton } from '../IconButton';
import readme from './README.md';

export default { title: 'Components/IconButton', component: IconButton, parameters: { readme } };

export const Basic = () => (
  <IconButton disabled={boolean('disabled', false)} onClick={action('onClick')}>
    <Icon icon="Close" />
  </IconButton>
);
