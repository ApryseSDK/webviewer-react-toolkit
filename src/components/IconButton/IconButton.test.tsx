import { mount, shallow } from 'enzyme';
import React from 'react';
import { spy } from 'sinon';
import { IconButton } from '../IconButton';

describe('IconButton component', () => {
  it('renders its contents', () => {
    const iconButton = shallow(<IconButton />);
    expect(iconButton.find('.ui__iconButton')).toHaveLength(1);
  });

  it('snapshot renders default iconButton', () => {
    const iconButton = shallow(<IconButton />);
    expect(iconButton).toMatchSnapshot();
  });

  it('clicking iconButton triggers onClick prop', () => {
    const onClick = spy();
    shallow(<IconButton onClick={onClick} />).simulate('click');
    expect(onClick.callCount).toBe(1);
  });

  it('clicking disabled iconButton does not trigger onClick prop', () => {
    const onClick = spy();
    // full DOM mount so `iconButton` element will use disabled prop
    mount(<IconButton onClick={onClick} disabled />).simulate('click');
    expect(onClick.callCount).toBe(0);
  });
});
