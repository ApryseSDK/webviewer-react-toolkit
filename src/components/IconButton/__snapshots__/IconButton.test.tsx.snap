// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component snapshot renders default iconButton 1`] = `
<ForwardRef
  buttonStyle="borderless"
  className="ui__base ui__iconButton"
/>
`;
